moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -133,21 +163,24 @@ class PlanningComponent
 
   /** \brief Run a plan from start or current state to fulfill the last goal constraints provided by setGoal() using
    * default parameters. */
-  bool plan();
+  MoveItErrorCode plan();
   /** \brief Run a plan from start or current state to fulfill the last goal constraints provided by setGoal() using the
    * provided PlanRequestParameters. */
-  bool plan(const PlanRequestParameters& parameters);
+  MoveItErrorCode plan(const PlanRequestParameters& parameters);
 
   /** \brief Execute the latest computed solution trajectory computed by plan(). By default this function terminates
    * after the execution is complete. The execution can be run in background by setting blocking to false. */
   bool execute(bool blocking = true);
 
+  /** \brief Return the last plan solution*/
+  PlanSolutionPtr getLastPlanSolution();
```

Comment:
I think this should return a const pointer to not allow changing the solution and then running `execute()` on it. Or at least make this const and add a separate function `getLastPlanSolutionNonConst()`.